app/src/main/java/org/simple/clinic/summary/PatientSummaryEffectHandler.kt

```diff
@@ -102,63 +104,69 @@ class PatientSummaryEffectHandler @AssistedInject constructor(
-                defaultHistoryUuid = uuidGenerator.v4(),
-                patientUuid = patient.uuid
-            )
-
-            StatinPrescriptionCheckInfoLoaded(
-                age = patient.ageDetails.estimateAge(userClock = userClock),
-                isPatientDead = patient.status == PatientStatus.Dead,
-                hasBPRecordedToday = hasBPRecordedToday,
-                assignedFacility = assignedFacility,
-                medicalHistory = medicalHistory,
-                prescriptions = prescriptions
-            )
-          }
-    }
-  }
-
-  private fun loadCVDRisk(): ObservableTransformer<LoadCVDRisk, PatientSummaryEvent> {
-    return ObservableTransformer { effects ->
-      effects
-          .observeOn(schedulersProvider.io())
-          .map {
-            val cvdRisk = cvdRiskRepository.getCVDRiskImmediate(it.patientUuid)
-            CVDRiskLoaded(cvdRisk?.riskScore)
+          .flatMap {
+            patientRepository.patient(it.patientUuid)
+                .extractIfPresent()
+                .flatMap { patient ->
+                  val cvdRisk = cvdRiskRepository.getCVDRiskImmediate(it.patientUuid)
+                  val today = LocalDate.now(userClock)
+                      .atStartOfDay(userClock.zone)
+                      .toInstant()
+
+                  Observable.combineLatest(
+                      medicalHistoryRepository.hasMedicalHistoryForPatientChangedSince(
+                          patientUuid = patient.uuid,
+                          instant = cvdRisk?.timestamps?.updatedAt ?: today,
+                      ),
+                      prescriptionRepository.newestPrescriptionsForPatient(patient.uuid),
+                      bloodPressureRepository.newestMeasurementsForPatient(
+                          patientUuid = patient.uuid,
+                          limit = 1
+                      ),
+                  ) { hasMedicalHistoryChanged, prescriptions, newestBp ->
+                    val patientAttribute = patientAttributeRepository.getPatientAttributeImmediate(
+                        patientUuid = patient.uuid
+                    )
+                    val medicalHistory = medicalHistoryRepository.historyForPatientOrDefaultImmediate(
+                        defaultHistoryUuid = uuidGenerator.v4(),
+                        patientUuid = patient.uuid
+                    )
+
+                    val ninetyDaysAgo = LocalDate.now(userClock)
+                        .minusDays(90)
+                        .atStartOfDay(userClock.zone)
+                        .toInstant()
+                    val wasBPMeasuredWithin90Days = newestBp.firstOrNull()?.updatedAt?.let { updatedAt ->
+                      updatedAt > ninetyDaysAgo
+                    } ?: false
+
+                    val wasCVDCalculatedWithin90Days = cvdRisk?.let { risk ->
+                      risk.timestamps.updatedAt > ninetyDaysAgo
```

Comment:
Same comment as above regarding the condition

Reply:
same comment as above